docs-chef-io/content/server/mutual_tls_errors.md

```diff
@@ -0,0 +1,309 @@
+distinguished_name  = req_distinguished_name
+string_mask         = utf8only
+
+# SHA-1 is deprecated, so use SHA-2 instead.
+default_md          = sha256
+
+# Extension to add when the -x509 option is used.
+x509_extensions     = v3_ca
+
+[ req_distinguished_name ]
+# See <https://en.wikipedia.org/wiki/Certificate_signing_request>.
+countryName                     = Country Name (2 letter code)
+stateOrProvinceName             = State or Province Name
+localityName                    = Locality Name
+0.organizationName              = Organization Name
+organizationalUnitName          = Organizational Unit Name
+commonName                      = Common Name
+emailAddress                    = Email Address
+
+# Optionally, specify some defaults.
+countryName_default             = YOUR_COUNTRY_GOES_HERE ex. US
+stateOrProvinceName_default     = YOUR_STATE_GOES_HERE ex. WA
+localityName_default            = YOUR_CITY_GOES_HERE ex. Seattle
+0.organizationName_default      = YOUR_BUSINESS_NAME ex. Friedlanders Hosiery
+organizationalUnitName_default  = YOUR_ORG_UNIT_NAME ex. Information Technology
+emailAddress_default            = YOUR_EMAIL ex. webmaster@friedlandershosiery.com
+
+# Subject Alternative Names - required for locally created certs
+[ alt_names ]
+DNS.1 = www.friedlandershosiery.com
+DNS.2 = friedlandershosiery.com
+DNS.3 = chef.friedlandershosiery.com
+
+[ v3_ca ]
+# Extensions for a typical CA (`man x509v3_config`).
+subjectKeyIdentifier = hash
+authorityKeyIdentifier = keyid:always,issuer
+basicConstraints = critical, CA:true
+keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+
+[ v3_intermediate_ca ]
+# Extensions for a typical intermediate CA (`man x509v3_config`).
+subjectKeyIdentifier = hash
+authorityKeyIdentifier = keyid:always,issuer
+basicConstraints = critical, CA:true, pathlen:0
+keyUsage = critical, digitalSignature, cRLSign, keyCertSign
+
+[ usr_cert ]
+# Extensions for client certificates (`man x509v3_config`).
+basicConstraints = CA:FALSE
+nsCertType = client, email
+nsComment = "OpenSSL Generated Client Certificate"
+subjectKeyIdentifier = hash
+authorityKeyIdentifier = keyid,issuer
+keyUsage = critical, nonRepudiation, digitalSignature, keyEncipherment
+extendedKeyUsage = clientAuth, emailProtection
+
+[ server_cert ]
+# Extensions for server certificates (`man x509v3_config`).
+basicConstraints = CA:FALSE
+nsCertType = server
+nsComment = "OpenSSL Generated Server Certificate"
+subjectKeyIdentifier = hash
+authorityKeyIdentifier = keyid,issuer:always
+keyUsage = critical, digitalSignature, keyEncipherment
+extendedKeyUsage = serverAuth
+subjectAltName = @alt_names
+
+[ crl_ext ]
+# Extension for CRLs (`man x509v3_config`).
+authorityKeyIdentifier=keyid:always
+
+[ ocsp ]
+# Extension for OCSP signing certificates (`man ocsp`).
+basicConstraints = CA:FALSE
+subjectKeyIdentifier = hash
+authorityKeyIdentifier = keyid,issuer
+keyUsage = critical, digitalSignature
+extendedKeyUsage = critical, OCSPSigning
+```
```

Comment:
In case of 
```
root@ip-10-0-12-174:~# chef-server-ctl user-list
ERROR: The data in your request was invalid
Response: <html>
 <head><title>400 The SSL certificate error</title></head>
 <body>
 <center><h1>400 Bad Request</h1></center>
 <center>The SSL certificate error</center>
 <hr><center>openresty/1.19.3.1</center>
 </body>
 </html>
```
Fetch certs:
```
root@ip-10-0-12-174:~# export PATH=/opt/opscode/embedded/bin:$PATH
root@ip-10-0-12-174:~# /opt/opscode/embedded/bin/knife ssl fetch -c /etc/opscode/pivotal.rb
```